### PR TITLE
BAEL-3658: Read actual cell value rather than the formula that is evaluating it

### PR DESCRIPTION
**Pull request by SunilJain95**
This PR contains code related to BAEL-3658. There are two ways to solve the problem so I have created two methods displaying one approach each.

I have also update Apache-POI version to the latest one.

### CODE REVIEW

File: apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        Workbook workbook = new XSSFWorkbook(inputStream);
+
+        Sheet sheet = workbook.getSheetAt(0);
+
+        CellAddress cellAddress = new CellAddress(cellLocation);
+        Row row = sheet.getRow(cellAddress.getRow());
+        Cell cell = row.getCell(cellAddress.getColumn());
+
+        if (cell.getCellType() == CellType.FORMULA) {
+            switch (cell.getCachedFormulaResultType()) {
```
Comment (eric-martin) [inline apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java]: Let's fix the indentation on the switch cases

File: apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        Sheet sheet = workbook.getSheetAt(0);
+        FormulaEvaluator evaluator = workbook.getCreationHelper()
+            .createFormulaEvaluator();
+
+        CellAddress cellAddress = new CellAddress(cellLocation);
+        Row row = sheet.getRow(cellAddress.getRow());
+        Cell cell = row.getCell(cellAddress.getColumn());
+
+        switch (evaluator.evaluateFormulaCell(cell)) {
+        case BOOLEAN:
```
Comment (eric-martin) [inline apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java]: let's fix the indentation on the switch cases here also

File: apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        Sheet sheet = workbook.getSheetAt(0);
+        FormulaEvaluator evaluator = workbook.getCreationHelper()
+            .createFormulaEvaluator();
+
+        CellAddress cellAddress = new CellAddress(cellLocation);
+        Row row = sheet.getRow(cellAddress.getRow());
+        Cell cell = row.getCell(cellAddress.getColumn());
+
+        switch (evaluator.evaluateFormulaCell(cell)) {
```
Comment (eric-martin) [inline apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java]: Lets add the same if statement that we have above for clarity
if (cell.getCellType() == CellType.FORMULA) {

File: apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java (hunk 1 above)
Comment (SunilJain95) [reply apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java]: Fixed the Indentation

File: apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java (hunk 2 above)
Comment (SunilJain95) [reply apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java]: Fixed the Indentation

File: apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java (hunk 3 above)
Comment (SunilJain95) [reply apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java]: Added the If condition for better visibility

**Pull request merged**